drivers/adc/ad9656/ad9656.c

```diff
@@ -0,0 +1,217 @@
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include "platform_drivers.h"
+#include "ad9656.h"
+
+/***************************************************************************//**
+ * @brief ad9656_spi_read
+ *******************************************************************************/
+int32_t ad9656_spi_read(struct ad9656_dev *dev,
+			uint16_t reg_addr,
+			uint8_t *reg_data)
+{
+	uint8_t buf[3];
+
+	int32_t ret;
+
+	// the MSB of byte 0 indicates a r/w operation, following 7 bits are the 
+	// bits 14-8 of the address of the register that is accessed. Byte 1 
+	// contains the bits 7-0 of the address of the register.
+	buf[0] = 0x80 | (reg_addr >> 8);
+	buf[1] = reg_addr & 0xFF;
+	buf[2] = 0x00;
+
+	ret = spi_write_and_read(dev->spi_desc,
+				 buf,
+				 3);
+	*reg_data = buf[2];
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief ad9656_spi_write
+ *******************************************************************************/
+int32_t ad9656_spi_write(struct ad9656_dev *dev,
+			 uint16_t reg_addr,
+			 uint8_t reg_data)
+{
+	uint8_t buf[3];
+
+	int32_t ret;
+
+	// the MSB of byte 0 indicates a r/w operation, following 7 bits are the 
+	// bits 14-8 of the address of the register that is accessed. Byte 1 
+	// contains the bits 7-0 of the address of the register. Byte 2 contains
+	// the data to be written.
+	buf[0] = reg_addr >> 8;
+	buf[1] = reg_addr & 0xFF;
+	buf[2] = reg_data;
+
+	ret = spi_write_and_read(dev->spi_desc,
+				 buf,
+				 3);
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief ad9656_setup
+ *******************************************************************************/
+int32_t ad9656_test(struct ad9656_dev *dev,
+		    uint32_t test_mode)
+{
+	ad9656_spi_write(dev,
+			 AD9656_REG_ADC_TEST_MODE,
+			 test_mode);
+	if (test_mode == AD9656_TEST_OFF)
+		ad9656_spi_write(dev,
+				 AD9656_REG_OUTPUT_MODE,
+				 AD9656_FORMAT_2S_COMPLEMENT);
+	else
+		ad9656_spi_write(dev,
+				 AD9656_REG_OUTPUT_MODE,
+				 AD9656_FORMAT_OFFSET_BINARY);
+	return(0);
+}
+
+/***************************************************************************//**
+ * @brief ad9656_setup
+ *******************************************************************************/
+int32_t ad9656_setup(struct ad9656_dev **device,
+		     const struct ad9656_init_param *init_param)
+{
+	uint8_t chip_id;
+	uint8_t pll_stat;
+	int32_t ret;
+	struct ad9656_dev *dev;
+
+	ret = 0;
```

Comment:
no need to initialize here if ret is used in read/write operations checks.

Reply:
Done